app/Services/AdventuresLeagueAdapter.php

```diff
@@ -83,33 +84,80 @@ public function getCharacter($filePath): ?Character
-                    'type' => Entry::TYPE_GAME,
+                    'event_id' => null,
+                    'dungeon_master_id' => null,
+                    'dungeon_master' => (array_key_exists(12, $data[$i])) ? (float)$data[$i][12] : "",
+                    'location' => (array_key_exists(11, $data[$i])) ? (float)$data[$i][11] : "",
+                    'levels' => 0,
+                    'date_played' => ($data[$i][3] == "") ? now() : $data[$i][3],
                     'gp' => (array_key_exists(7, $data[$i])) ? (float)$data[$i][7] : 0,
+                    'downtime' => (array_key_exists(8, $data[$i])) ? (float)$data[$i][8] : 0,
+                    'type' => $type,
                 ];
-                array_push($entries, new Entry($entryData));
-            }
-            // Item importing to be refactored
-            /*else {
-                //create the item and attach it to the last entry
-                $latestEntry = end($entries);
-                if (is_null($latestEntry)) {
-                    continue;
+                $entry = Entry::create($entryData);
+                $entry->save();
+                array_push($entries, $entry);
+            } elseif ($isItemEntry) {
+                if (count($entries) == 0) {
+                    $e = Entry::create([
+                        'user_id' => Auth::id(),
+                        'adventure_id' => null,
+                        'campaign_id' => null,
+                        'character_id' => $characterId,
+                        'event_id' => null,
+                        'dungeon_master_id' => null,
+                        'levels' => 0,
+                        'date_played' => now(),
+                        'gp' => 0,
+                        'downtime' => 0,
+                        'type' => Entry::TYPE_GAME,
+                    ]);
+                    $e->save();
+                    array_push($entries, $e);
                 }
+
+                $rarity = ((array_key_exists(2, $data[$i])) ? (string) $data[$i][2] : "common");
+                $lastEntry = $entries[array_key_last($entries)];
+
                 $itemData = [
-                    'name' => $data[$i][1],
-                    'rarity' => $data[$i][2],
+                    'entry_id' => $lastEntry->id,
+                    'character_id' => $characterId,
+                    'name' => (array_key_exists(1, $data[$i])) ? $data[$i][1] : "",
+                    'rarity' => in_array($rarity, Item::RARITY) ? $rarity : "uncommon", //need enum
+                    'description' => (array_key_exists(6, $data[$i])) ? $data[$i][6] : "",
+                    'author_id' => Auth::id(),
+                    'tier' => 0,
+                    'counted' => 0,
                 ];
-                $latestEntry->items()->save(new Item($itemData));
-            }*/
+                $item = new Item($itemData);
+                $item->save();
```

Comment:
same here i think this will be a prime candidate for method extractions next time we refactor :)